bubblewrap.c

```diff
@@ -1474,7 +1474,28 @@ setup_newroot (bool unshare_pid,
         case SETUP_MAKE_SYMLINK:
           assert (op->source != NULL);  /* guaranteed by the constructor */
           if (symlink (op->source, dest) != 0)
-            die_with_error ("Can't make symlink at %s", op->dest);
+            {
+              if (errno == EEXIST)
+                {
+                  char existing[PATH_MAX + 1] = "";
+                  ssize_t elen = readlink (dest, existing, sizeof (existing) - 1);
+                  if (elen < 0)
+                    {
+                      if (errno == EINVAL)
+                        die ("Can't make symlink at %s: destination exists and is not a symlink", op->dest);
+                      else
+                        die_with_error ("Can't make symlink at %s: destination exists, and cannot read symlink target", op->dest);
+                    }
+
+                  existing[elen] = '\0';
```

Comment:
Please use `readlink_malloc()` (into a new `cleanup_free char *existing = NULL`) in preference to a statically-sized buffer. This returns a non-null string on success, or `NULL` with `errno` set on failure.

Reply:
hey @jkaivo will you be able to do this chages?

Reply:
I'm just seeing this. I don't see what actual advantage `readlink_malloc()` has over POSIX `readlink()` with an appropriately sized buffer. Since submitting this patch, I have solved my personal use-case issues by only using Steam on my Steam Deck, so I have very little motivation to make additional changes.

Reply:
> I don't see what actual advantage readlink_malloc() has over POSIX readlink() with an appropriately sized buffer

If it was possible to choose an appropriately sized buffer, then there would be no advantage. The problem with POSIX readlink() is that there is no correct size for the buffer, because Linux `PATH_MAX` is a lie (it is entirely valid to have paths longer than `PATH_MAX`).